src/app/ipv/middleware.js

```diff
@@ -358,87 +341,54 @@ module.exports = {
       delete req.session.currentPageStatusCode;
     }
   },
+
+  // make considerations for handleEscapeAction pages with no req.body?.journey that defaults to `end`
   handleJourneyAction: async (req, res, next) => {
     const currentPageId = req.params.pageId;
+    const pagesUsingSessionId = [
+      "pyi-suggest-other-options",
+      "pyi-cri-escape",
+      "pyi-kbv-escape-m2b",
+      "pyi-escape-m2b",
+      "page-multiple-doc-check",
+    ];
+    const journeyActions = {
+      end: "end",
+      "address-current": "address-current",
+      "attempt-recovery": "attempt-recovery",
+      "build-client-oauth-response": "build-client-oauth-response",
+      next: "next",
+      "next/end": "end",
+      "next/f2f": "f2f",
+      "next/dcmaw": "dcmaw",
+      "next/bank-account": "bankAccount",
+      "next/passport": "ukPassport",
+      "next/driving-licence": "drivingLicence",
+    };
     try {
-      checkForIpvAndOauthSessionId(req, res);
-
-      if (req.body?.journey === "end") {
-        await handleJourneyResponse(req, res, "end", currentPageId);
-      } else if (req.body?.journey === "addressCurrent") {
-        await handleJourneyResponse(req, res, "address-current", currentPageId);
-      } else if (req.body?.journey === "attempt-recovery") {
-        await handleJourneyResponse(
-          req,
-          res,
-          "attempt-recovery",
-          currentPageId,
-        );
-      } else if (req.body?.journey === "build-client-oauth-response") {
-        req.session.ipAddress = req?.session?.ipAddress
-          ? req.session.ipAddress
-          : getIpAddress(req);
-        await handleJourneyResponse(
-          req,
-          res,
-          "build-client-oauth-response",
-          currentPageId,
-        );
-      } else {
-        await handleJourneyResponse(req, res, "next", currentPageId);
-      }
-    } catch (error) {
-      transformError(error, `error handling POST request on ${currentPageId}`);
-      next(error);
-    }
-  },
-
-  handleMultipleDocCheck: async (req, res, next, currentPageId) => {
-    try {
-      checkForSessionId(req, res);
-
-      if (req.body?.journey === "next/passport") {
-        await handleJourneyResponse(req, res, "ukPassport", currentPageId);
-      } else if (req.body?.journey === "next/driving-licence") {
-        await handleJourneyResponse(req, res, "drivingLicence", currentPageId);
+      const action = journeyActions[req.body?.journey] || "next";
+      if (pagesUsingSessionId.includes(currentPageId)) {
+        checkForSessionId(req, res);
       } else {
-        await handleJourneyResponse(req, res, "end", currentPageId);
+        checkForIpvAndOauthSessionId(req, res);
```

Comment:
I didn't look that deep into it. But on the surface I figure that some pages **require** only the ipvSession ID while some **require** both the ipvSession and the OauthSession ID. And they fail if any of the required session ID is not present. 

I am not exactly sure why, as I think by default journeys should have both IDs present at each stage. 

I would look deeper and see what the reasoning behind the differentiation is and if changing it would break any test

Reply:
I think this method looks for the `ipvSessionId` OR `clientOauthSessionId` (not AND), so I was hoping we could just use that for everything.

It's fine to leave, since you've not changed the logic, but I suspect it isn't quite right.